File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    # skip 내의 문자를 제외한 알파벳 리스트 생성
    alpha = [chr(i) for i in range(97, 123) if not chr(i) in skip] * 3 
    
    # s 내의 문자를 가져와 index 뒤의 알파벳으로 변경한 문자를 리스트로 저장 후 join 반환
    return ''.join([alpha[alpha.index(i) + index] for i in s])

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>

using namespace std;

bool check(char c, string skip) {
    for (int i=0; i<skip.size(); i++) {
        if (c == skip[i]) return false;
    }
    return true;
}

string solution(string s, string skip, int index) {
    string ret = "";

    for (char c: s) {
        for (int i=0; i<index;) {
            c += 1;
            if (c == 'z'+1) c = 'a';
            if (check(c, skip)) i++;
        }
        ret += c;
    }
    return ret;
}